### ISSUE TITLE
Proposed changes to rstar defaults

### ISSUE THREAD

**Issue by sethaxen** [TuringLang/MCMCDiagnosticTools.jl#51]
The defaults of our `rstar` differ slightly from the paper:
- while the paper recommends a 70-30 split of train-test data, we default to 80-20
- the paper describes the split as happening within each chain, i.e. it is stratified by chains, while we split all draws
- while the paper recommends splitting each chain in two to diagnose poor within-chain convergence as well, we do not offer this as an option (it would be straightforward with #50)

I propose that with the breaking changes in #50 we update `rstar` to:
- use chain-stratified splitting with an ~80-20~ 70-30 split by default
- add a keyword `split_chains`, defaulting to `true`.

**Comment by sethaxen** [comment]
@devmotion what do you of including these changes in the same breaking release as #50?

**Comment by devmotion** [comment]
> use chain-stratified splitting with an 80-20 split by default

So keeping the 80-20 and not switching to the 70-30 split recommended in the paper?

Generally, I must admit I haven't read the paper in detail as the first implementation was added by @trappmartin in https://github.com/TuringLang/MCMCChains.jl/pull/238 (IIRC I mainly did some refactoring and replaced the sampling step with the analytical Poisson-Binomial distribution in https://github.com/TuringLang/MCMCDiagnosticTools.jl/pull/7). It seems already in the first implementation we used a 80-20 split. I think it would be good to follow the paper closely if there is no good evidence that a different choice would be better.

I would view these changes as bug fixes and hence non-breaking (since they fix inconsistencies with the paper). Therefore I think they do not have to be included in the same breaking release as #50. In any case, it would probably good to open a separate PR and not add them to #50.

**Comment by sethaxen** [comment]
> So keeping the 80-20 and not switching to the 70-30 split recommended in the paper?

Ah, sorry, I mixed them up. Fixed. No, I'm suggesting switching to the paper's.

> I would view these changes as bug fixes and hence non-breaking (since they fix inconsistencies with the paper). Therefore I think they do not have to be included in the same breaking release as #50. In any case, it would probably good to open a separate PR and not add them to #50.

Yes these could be viewed as bug fixes. If we go with that interpretation, then sure I can add them in a patch release after #50 is merged. I agree, they should not be in #50 itself.
